#! /usr/bin/env python
###########################################################################
# Copyright (C) 2017-2018 FEUP-LSTS - www.lsts.pt                         #
###########################################################################
# This program is free software; you can redistribute it and/or modify    #
# it under the terms of the GNU General Public License as published by    #
# the Free Software Foundation; either version 2 of the License, or (at   #
# your option) any later version.                                         #
#                                                                         #
# This program is distributed in the hope that it will be useful, but     #
# WITHOUT ANY WARRANTY; without even the implied warranty of              #
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU        #
# General Public License for more details.                                #
#                                                                         #
# You should have received a copy of the GNU General Public License       #
# along with this program; if not, write to the Free Software             #
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA           #
# 02110-1301 USA.                                                         #
###########################################################################
# Author: Paulo Dias                                                      #
###########################################################################

PACKAGE = "ros_imc_broker"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("enable_loopback", bool_t, 0, "If loopback should be ignored ", False)

gen.add("system_name", str_t, 0, "System name", "turbot")
gen.add("system_type", str_t, 0, "System type", "uuv")
gen.add("system_imc_id", int_t, 0, "IMC ID", 0x2000, 0, 0xFFFF)

gen.add("udp_port", int_t, 0, "Server's UDP port" , 6002, 0)
gen.add("udp_port_tries", int_t, 0, "Server's UDP port tries" , 10, 1, 20)
gen.add("udp_contact_timeout", int_t, 0, "UDP connect timeout in seconds" , 30, 1, 300)

gen.add("multicast_addr", str_t, 0, "Multicast group address", "224.0.75.69")
gen.add("multicast_port", int_t, 0, "Multicast port", 30100, 0)
gen.add("multicast_port_range", int_t, 0, "Multicast port range", 5, 1, 10)

gen.add("additional_services", str_t, 0, "Additional services URI to announce", "")

gen.add("static_destinations_addrs", str_t, 0, "Comma separated IP:port destinations", "")

exit(gen.generate(PACKAGE, "ros_imc_broker", "AdapterParams"))
